File: Ejercicios Propuestos 3/Ejercicio 7/Ejercicio 7.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Solicitar la cantidad de números
n = int(input("Ingrese la cantidad de números: "))

# Crear una lista vacía para almacenar los números
vector = []

# Leer los números del usuario
for i in range(n):
    numero = int(input(f"Número para la posición {i}: "))
    vector.append(numero)

# Insertion Sort
for i in range(1, n):
    # Guardar el elemento actual
    key = vector[i]

    # Inicializar el índice del elemento anterior
    j = i - 1

    # Desplazar hacia la derecha los elementos mayores que 'key'
    while j >= 0 and vector[j] > key:
        vector[j + 1] = vector[j]
        j -= 1

    # Insertar 'key' en su posición correcta
    vector[j + 1] = key

# Mostrar el vector ordenado
print("Vector ordenado:")

for numero in vector:
    print(numero, end=" ")

print()  # Salto de línea

File: Ejercicios Propuestos 3/Ejercicio 7/ejercicio7.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Ingrese la cantidad de elementos: ";
    cin >> n;

    int v[100];
    for (int i = 0; i < n; i++) {
        cout << "Elemento " << i << ": ";
        cin >> v[i];
    }

    for (int i = 1; i < n; i++) {
        int clave = v[i];
        int j = i - 1;
        while (j >= 0 && v[j] > clave) {
            v[j + 1] = v[j];
            j = j - 1;
        }
        v[j + 1] = clave;
    }

    cout << "Vector ordenado: ";
    for (int i = 0; i < n; i++) {
        cout << v[i] << " ";
    }
    cout << endl;
    return 0;
}
